components/net/image_cache.rs

```diff
@@ -551,6 +711,8 @@ impl ImageCache for ImageCacheImpl {
             .unwrap_or(pref!(threadpools_fallback_worker_num) as usize)
             .min(pref!(threadpools_async_runtime_workers_max).max(1) as usize);
 
+        // The PipelineId will be None because it belongs to the script_thread itself.
+        // Later ImageCaches use `create_new_image_cache` to get create an image cache for a given pipeline.
```

Comment:
I would add this direction above `pipeline_id`, and reword it as: "The constructed cache is not associated with any pipeline yet, which will happen later by way of `create_new_image_cache`".